external-applications.md: Add section on Steam (#496)
void-linux/void-docs: src/config/external-applications.md

To use MATLAB's help browser, live scripts, add-on installer, and simulink,
install the `libselinux` package.

## Steam

Steam can be installed either via a native package, which requires [enabling the
"nonfree" repository](../xbps/repositories/index.md#nonfree), or via

> [Flatpak](##flatpak). The list of dependencies for different platforms and

Reviewer: ```suggestion
[Flatpak](#flatpak). The list of dependencies for different platforms and
```

troubleshooting information for the native package can be found in its
[Void-specific documentation](./package-documentation/index.html), while this

> section will deals with potential issues faced by Flatpak users.

Reviewer: ```suggestion
section will deal with potential issues faced by Flatpak users.
```

> If you are using a different drive to store you game library, the `--filesystem`

Reviewer: ```suggestion
If you are using a different drive to store your game library, the `--filesystem`
```

option from [flatpak-override(1)](https://man.voidlinux.org/flatpak-override.1)
can prove useful.

If sound is not working when Steam is installed using Flatpak,
[PulseAudio](./media/pulseaudio.md) auto-activation might not be working

> correctly. Make sure PulseAudio is running before launching Steam.

Reviewer: I think we can consider moving this to the flatpak item, since other flatpak applications have the same issue. What do you think?

Author: I think it looks better in the flatpak section and as such I have moved it there in my most recent push.

To use MATLAB's help browser, live scripts, add-on installer, and simulink,
install the `libselinux` package.

## Steam

Steam can be installed either via a native package, which requires [enabling the
"nonfree" repository](../xbps/repositories/index.md###nonfree), or via
[Flatpak](##flatpak). Dependency listings for different platforms and
troubleshooting information for the native package can be found in its
[Void-specific documentation file](./package-documentation/index.html), so the
remainder of this section will deal with potential issues faced by Flatpak
users.

> If you are using a different drive to store you game library run the following
> commmand, where `<storage>` is the mount point of your drive:
> 
> `$ flatpak override com.valvesoftware.Steam --filesystem=<storage>`

Reviewer: ```suggestion
If you are using a different drive to store you game library, the `--filesystem` option from [flatpak-override(1)](https://man.voidlinux.org/flatpak-override.1) can prove useful.
```

If you think an example command is really important, it should be a full code block, using three \`\`\`.

Author: No, I think your suggestion with a link to the man page is better. I'll include it in my next push.

> If sound is not working when Steam is installed in this manner, creating and
> running the following script to start Steam may fix the issue:
> 
> ```
> #!/bin/sh
> pulseaudio --start flatpak run com.valvesoftware.Steam
> ```

Reviewer: ```suggestion
If sound is not working when Steam is installed with Flatpak, [PulseAudio](link to pulse) auto-activation might not be working correctly. Make sure PulseAudio is running before launching Steam.
```

Author: Do you think this advice is complete enough to get rid of the script that was included in the wiki? I'm not super familiar with pulse auto-activation, but my initial assumption is that it would be useful to keep the script (or maybe just the command) for users whose audio is not working even with pulse installed.

Reviewer: I think if we recommend this at all, it should go in the PulseAudio section. But I think I'd rather see a recommendation for launching a terminal and running `pulseaudio --daemonize=no --exit-idle-time=-1` or something of the sort.

Author: I understand. I'll see about adding something like that to the PulseAudio page.

Author: Just opened a PR: #565.

To use MATLAB's help browser, live scripts, add-on installer, and simulink,
install the `libselinux` package.

## Steam

### Installing Steam through Flatpak

Steam can be most easily be installed via [Flatpak](##flatpak). If you are using
a different drive to store you game library run the following commmand, where
`<storage>` is the mount point of your drive:

`flatpak override com.valvesoftware.Steam --filesystem=<storage>`

If sound is not working when Steam is installed via Flatpak, creating and
running the following script to start Steam may fix the issue:

```
#!/bin/sh
pulseaudio --start flatpak run com.valvesoftware.Steam
```

### Installing Steam through XBPS

To install Steam you first need to [enable the "nonfree"
repository](../xbps/repositories/index.md###nonfree). Then, install the `steam`
package; this should be enough for i686 systems. On x86_64 systems, additional
32-bit libraries must be installed. These libraries are listed upon installing
the `steam` package, and can be viewed in [Steam's install
message](https://github.com/void-linux/void-packages/blob/master/srcpkgs/steam/INSTALL.msg).

> You will need to [enable](../config/services/index.md####enabling-services) the

Reviewer: Kind of unrelated, but I think we could add a tiny item to the XBPS part about seeing install msgs (`xbps-query -p install-msg <package>`), and this would link there instead of to the git repo.
